### ISSUE TITLE
carthage build fails: macOSX 10.11+ sdk features used, deployment target 10.10

### ISSUE THREAD

**Issue by mfogel** [auth0/Auth0.swift#269]
### Description

Running `carthage bootstrap` fails to compile for scheme `Auth0.macOS`. The following error appears in the build log:

```
Carthage/Checkouts/Auth0.swift/Auth0/CredentialsManager.swift:79:53: error: 'archivedData(withRootObject:)' is only available on OS X 10.11 or newer
        return self.storage.setData(NSKeyedArchiver.archivedData(withRootObject: credentials), forKey: storeKey)
```

Some of the deployment targets in `Auth0.swift/Auth0.xcodeproj/project.pbxproj` appear to be set to `MACOSX_DEPLOYMENT_TARGET = 10.10`. If I change them all to `MACOSX_DEPLOYMENT_TARGET = 10.11` then the build goes through fine.

### Prerequisites

* [X] Did you check the documentation? [Add a link]
* [X] Did you check the [Auth0 Community](https://community.auth0.com/)?
* [X] Are you reporting this to the correct repository?
* [X] Are there any related or duplicate Issues or PRs for this issue?

### Environment

Please provide the following:

- Auth0.swift version `1.14.2`
- iOS version
- If using Carthage or CocoaPods, please include your Cartfile or Podfile

```
github "auth0/Auth0.swift" ~> 1.14
github "auth0/JWTDecode.swift" ~> 2.2
```

- Xcode version `10.2`

### Reproduction

```
mfogel@BA:~/ $ mkdir tmp
mfogel@BA:~/ $ cd tmp
mfogel@BA:~/tmp/ $ echo 'github "auth0/Auth0.swift" ~> 1.14' > Cartfile
mfogel@BA:~/tmp/ $ carthage bootstrap
*** No Cartfile.resolved found, updating dependencies
*** Fetching Auth0.swift
*** Fetching SimpleKeychain
*** Checking out Auth0.swift at "1.14.2"
*** Checking out SimpleKeychain at "0.8.1"
*** xcodebuild output can be found in /var/folders/hn/1m5q4ldj7rggq1v2p69f1jwh0000gn/T/carthage-xcodebuild.HjD4ZO.log
*** Building scheme "SimpleKeychain-OSX" in SimpleKeychain.xcodeproj
*** Building scheme "SimpleKeychain-iOS" in SimpleKeychain.xcodeproj
*** Building scheme "SimpleKeychain-watchOS" in SimpleKeychain.xcodeproj
*** Building scheme "SimpleKeychain-tvOS" in SimpleKeychain.xcodeproj
*** Building scheme "Auth0.tvOS" in Auth0.xcodeproj
*** Building scheme "Auth0.macOS" in Auth0.xcodeproj
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -project /Users/mfogel/tmp/Carthage/Checkouts/Auth0.swift/Auth0.xcodeproj -scheme Auth0.macOS -configuration Release -derivedDataPath /Users/mfogel/Library/Caches/org.carthage.CarthageKit/DerivedData/10.2_10E125/Auth0.swift/1.14.2 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/hn/1m5q4ldj7rggq1v2p69f1jwh0000gn/T/Auth0.swift SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/mfogel/tmp/Carthage/Checkouts/Auth0.swift)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/hn/1m5q4ldj7rggq1v2p69f1jwh0000gn/T/carthage-xcodebuild.HjD4ZO.log
```

**Comment by cocojoe** [comment]
What version of Carthage? I didn't have a problem building in Xcode 10.2.

**Comment by mfogel** [comment]
Carthage version 0.33.0

**Comment by tipok** [comment]
I'm getting the same issue using Cocoapods in AppCode 2018.3.6.
After changing `MACOSX_DEPLOYMENT_TARGET` as described in the issue description, builds fine.

**Comment by VictorNouvellet** [comment]
I think it is related to this radar: https://openradar.appspot.com/49262697

**Comment by cocojoe** [comment]
Thanks, I've just bumped min version in #272 although not sure on this, is technically a BC. However, 10.10 is very old. 

Wondering if the radar issues is fixed in Xcode 10.2.1 as this was not an issue in Xcode 10.1

**Comment by mfogel** [comment]
This is fixed for me now with v1.15.0
